### ISSUE TITLE
Add getValueOrDefault on Map

### ISSUE THREAD

**Issue by lucasteles** [fsharp/fslang-suggestions#1125]
This is a common operation in dictionaries, would be nice to have a way to get a value from it, if the key does not exist return a default value.

The `Map` type already has an implementation of this for `IReadOnlyDictionary`.  I think this should be in the default methods and API for `Map`

Implementation:
```fsharp
let getOrDefault k default' map = 
    (map :> IReadOnlyDictionary<_,_>).GetValueOrDefault(k, default')


let map = Map([("a",1)])
map |> getOrDefault "a" 0  // 1
map |> getOrDefault "b" 0  // 0

```

**Comment by abelbraaksma** [comment]
This is a copy of https://github.com/dotnet/fsharp/issues/12884, moved here because it is a language suggestion. Just linking back so googlers can find it ;).

**Comment by abelbraaksma** [comment]
If I understand your suggesting correctly, this is basically `Map.tryFind` combined with `Option.defaultValue`, right? I.e.:

```f#
let getValueOrDefault defaultValue key =
    Map.tryFind key >> Option.defaultValue defaultValue
```

Using it:

```f#
> let x = Map [("foo", 42)];;
val x: Map<string,int> = map [("foo", 42)]

> x |> getValueOrDefault 99 "bar";;
val it: int = 99

> x |> getValueOrDefault 99 "foo";;
val it: int = 42
```

**Comment by lucasteles** [comment]
@abelbraaksma exactly

Today we can have it as an extension on [System.Collections.Generic](https://github.com/dotnet/runtime/blob/bdbdaa8e2b4159c6839ca1c65b280fc25b0bbb82/src/libraries/System.Collections/src/System/Collections/Generic/CollectionExtensions.cs#L16) too

Is something like:
```fsharp
let getOrDefault key default' (map: Map<_,_>) =
    match map.TryGetValue key with
    | true, value -> value
    | _ -> default'
```

**Comment by akhansari** [comment]
I believe it's usually a bad practice to have `getValueOrDefault`.
If there is no value so then it's None.
I saw too much bugs in C# related to this design and I'm happy that there are not any in F#.

**Comment by lucasteles** [comment]
@akhansari I don't think this is true If the default value is explicitly passed to the function

**Comment by akhansari** [comment]
@lucasteles My bad, I didn't see that the default value is explicitly passed to the function. All good as long as everything is explicit 👍🏻
